packages/safe-tools-client/app/components/token-to-usd/index.gts

```diff
@@ -19,26 +21,28 @@ const INTERVAL = config.environment === 'test' ? 1000 : 60 * 1000;
 
 export default class TokenToUsd extends Component<Signature> {
   @service('token-to-usd') declare tokenToUsdService: TokenToUsdService;
+  @service declare tokens: TokensService;
   updateInterval: ReturnType<typeof setInterval>;
 
   constructor(owner: unknown, args: Args) {
     super(owner, args);
-
     this.updateInterval = setInterval(() => {
       taskFor(this.tokenToUsdService.updateUsdConverter).perform(args.tokenAddress); 
     }, INTERVAL);
   }
   
   get usdAmount() {
-    return this.tokenToUsdService.toUsd(this.args.tokenAddress, BigNumber.from(this.args.tokenAmount))?.toString();
+    return this.tokenToUsdService.toUsd(this.args.tokenAddress, BigNumber.from(this.args.tokenAmount));
   }
 
   willDestroy() {
     clearInterval(this.updateInterval);
   }
 
   <template>
-    {{this.usdAmount}}
+    {{#if this.usdAmount}}
+      $ {{(weiToDecimal this.usdAmount 18)}} USD
```

Comment:
Is 18 correct in all cases? Don't some tokens have 6 decimals?

Reply:
Sorry, I pushed the wrong code. Updated.